### PR TITLE
Linked List implementation for finding the middle element.

### PR DESCRIPTION
**Pull request by maximilian-mathers**
#### Description of Change
<--
This pull request is for adding a feature, Tortoise and Hare algorithm implemented with a linked list to find the middle element, into the C++ version of TheAlgorithms. The commit includes a new file titled "linked_list_find_middle.cpp" which is the only section changes were made to.

Contributors guide: https://github.com/TheAlgorithms/C-Plus-Plus/CONTRIBUTING.md
-->

#### Checklist

- [x ] Added description of change
- [ x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x ] Added tests and example, test must pass
- [x ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x ] Relevant documentation/comments is changed or added
- [x ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x ] I acknowledge that all my contributions will be made under the project's license.

Notes: Added linked list implementation of the Tortoise and Hare algorithm to find the middle element of a linked list.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1923"></a>

### CODE REVIEW

File: data_structures/linked_list_find_middle.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    //the middle element of the linked list
+    while(fastptr && fastptr->next && fastptr->next->next)
+    {
+        //move by 1 for slowptr
+        slowptr = slowptr->next;
+        //move by 2 for fastptr
+        fastptr = fastptr->next->next;
+    }
+    //store value of middle node and return the integer value
+    int ans = slowptr->val;
```
Comment (Panquesito7) [inline data_structures/linked_list_find_middle.cpp:66]: Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

File: data_structures/linked_list_find_middle.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+    
+    printArr(n, numNodes);
+    //destroys dynamically allocated array
+    delete[] n;
+    int midElement = midPoint(&head);
+    std::cout<<std::endl;
+    std::cout<< "The value of the middle element is: " << midElement << std::endl;
+    return 0;
+}
```
Comment (Panquesito7) [inline data_structures/linked_list_find_middle.cpp:149]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 😄
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/maximilian-mathers/C-Plus-Plus/actions

File: data_structures/linked_list_find_middle.cpp (hunk 1 above)
Comment (maximilian-mathers) [reply data_structures/linked_list_find_middle.cpp]: I will use the uint32_t in my next commit and specify to users, let me know if it checks out.

**Pull request closed without merging**